x-pack/plugins/ml/public/application/components/custom_urls/custom_url_editor/editor.tsx

```diff
@@ -360,66 +374,82 @@ export const CustomUrlEditor: FC<CustomUrlEditorProps> = ({
               />
             </EuiFormRow>
           )}
-        {type === URL_TYPE.KIBANA_DASHBOARD ||
-        (type === URL_TYPE.KIBANA_DISCOVER && showCustomTimeRangeSelector && hasTimefield) ? (
-          <CustomTimeRangePicker
-            onCustomTimeRangeChange={onCustomTimeRangeChange}
-            customTimeRange={customUrl?.customTimeRange}
-          />
-        ) : null}
 
-        {(type === URL_TYPE.KIBANA_DASHBOARD || type === URL_TYPE.KIBANA_DISCOVER) &&
-          showTimeRangeSelector && (
-            <>
-              <EuiSpacer size="m" />
-              <EuiFlexGroup>
-                <EuiFlexItem grow={false}>
-                  <EuiFormRow
-                    label={
-                      <FormattedMessage
-                        id="xpack.ml.customUrlsEditor.timeRangeLabel"
-                        defaultMessage="Time range"
-                      />
-                    }
-                    className="url-time-range"
-                    display="rowCompressed"
-                  >
-                    <EuiSelect
-                      options={timeRangeOptions}
-                      value={timeRange.type}
-                      onChange={onTimeRangeTypeChange}
-                      data-test-subj="mlJobCustomUrlTimeRangeInput"
-                      compressed
-                    />
-                  </EuiFormRow>
-                </EuiFlexItem>
-                {timeRange.type === TIME_RANGE_TYPE.INTERVAL && (
-                  <EuiFlexItem>
+        {(type === URL_TYPE.KIBANA_DASHBOARD || type === URL_TYPE.KIBANA_DISCOVER) && (
+          <>
+            <EuiSpacer size="m" />
+            <EuiFlexGroup direction="column" gutterSize="xs">
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup alignItems="center">
+                  <EuiFlexItem grow={false}>
                     <EuiFormRow
                       label={
                         <FormattedMessage
-                          id="xpack.ml.customUrlsEditor.intervalLabel"
-                          defaultMessage="Interval"
+                          id="xpack.ml.customUrlsEditor.timeRangeLabel"
+                          defaultMessage="Time range"
                         />
                       }
                       className="url-time-range"
-                      error={invalidIntervalError}
-                      isInvalid={isInvalidTimeRange}
                       display="rowCompressed"
                     >
-                      <EuiFieldText
-                        value={timeRange.interval}
-                        onChange={onTimeRangeIntervalChange}
-                        isInvalid={isInvalidTimeRange}
-                        data-test-subj="mlJobCustomUrlTimeRangeIntervalInput"
-                        compressed
-                      />
+                      {showCustomTimeRangeSelector && hasTimefield ? (
```

Comment:
The time range interval dropdown should be hidden for jobs where there is no time field in the results. 

So here, using `iris`, for Dashboard type URLs only the 'Add custom time range' option should be available.

<img width="333" alt="image" src="https://github.com/elastic/kibana/assets/7405507/3f8f4437-3ae5-4e7e-8b6b-763f4f4a4b1c">

And with `iris` for Discover type URLs, no time range options should be available:

<img width="391" alt="image" src="https://github.com/elastic/kibana/assets/7405507/27d923ce-e8c3-47b4-97a9-200c74cc81ea">


The interval option doesn't apply here I think as there is no time field to apply it to in the results.

Reply:
Good catch - fixed in 124d047b2f4b880c65e2cec0bbd980421b74b4bd